pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -504,9 +504,10 @@ private void uploadSegment(File indexDir, SegmentMetadata segmentMetadata, File
     // Brand new segment, not refresh, directly add the segment
     if (znRecord == null) {
       LOGGER.info("Adding new segment: {}", segmentName);
-      if (downloadUrl == null) {
+      // We will only assume that segment metadata upload doesn't have to do a move through the controller because
```

Comment:
Why does segment metadata upload imply that the segment is in its final location? These two seem unrelated, and should not be tied together. If that is because controller moving the segment requires segment to download, then we should revisit why that is required and try to fix that, instead of building on top of that.